GitGuide.md

```diff
@@ -0,0 +1,44 @@
+# Git Guide 
+## Prerequisite
+1. At the heart of GitHub is an open source version control system (VCS) called Git. Git is responsible for everything GitHub-related that happens locally on your computer. 
+   1. [Download and install the latest version of GIT](https://git-scm.com/downloads)
+   2. [Set your username in GIT](https://docs.github.com/en/get-started/getting-started-with-git/setting-your-username-in-git)
+   3. [Set your commit email address in Git](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-email-preferences/setting-your-commit-email-address)
+   4. [Authenticating with GitHub from Git](https://docs.github.com/en/get-started/quickstart/set-up-git#authenticating-with-github-from-git)
+   5. [Add SSH keys in GITHUB](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/about-ssh)
+2. It good to have some git desktop client to work fast. it is optional and totally depends upon the individual. Some recommendation are list down
+   1. [Github Desktop](https://desktop.github.com/) 
+   2. [Source Tree](https://www.sourcetreeapp.com/)
+   3. [Tortoise Git](https://tortoisegit.org/)
+## GitHub Flow Overview 
+   1. clone repository from GitHub 
+   2. create a working branch on local computer
+   3. Implement the requested feature, make sure all tests are passing, and commit all changes in the new branch
+   4. checkout the main branch locally 
+   5. Merge main branch into your local branch    
+   6. Resolve merge conflicts if occurs
+   7. push/upload the changes to remote branch

```

Comment:
Isn't this flow now pushing straight to main branch? Should it be something like, clone repo, create new branch, implement stuff, commit and push to remote, create pull request, resolve merge conflicts. https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/addressing-merge-conflicts/resolving-a-merge-conflict-on-github here is a good link how you can use github to resolve conflicts.

Reply:
no it is not pushing to main branch. i have updated the line 7.